src/main/java/edu/hawaii/its/api/service/MembershipService.java

```diff
@@ -92,6 +109,11 @@ private List<MembershipResult> createMemberships(List<Group> membershipGroupings
             membershipResult.setDescription(grouping.getDescription());
             membershipResult.setPath(grouping.getGroupPath());
             membershipResult.setName(nameGroupingPath(grouping.getGroupPath()));
+            GroupAttributeResults groupingAttributeResults =
+                    grouperService.groupAttributeResults(CURATED, grouping.getGroupPath());
+            if (groupingAttributeResults.getResultMessage().equals(", Found 1 results.")) {
+                membershipResult.setName(grouping.getGroupPath());
+            }
```

Comment:
Instead of having to make an additional call to Grouper to check which is a curatedGrouping, you could also pass in the list of curatedGroupings as a parameter to this `createMemberships` function. Also I think the MembershipResult object deserves a new constructor where you can pass the name, path, and description into it.

Reply:
Making a call to grouper to check whether every membership is a curated grouping can be slow, which is why I'm suggesting the above. Since you already know which are the curated groupings, you can just pass it into the `createMemberships` function.